src/python/pants/engine/internals/graph.py

```diff
@@ -255,6 +256,91 @@ async def resolve_all_generator_target_requests(
     )
 
 
+async def _parametrized_target_generators_with_templates(
+    address: Address,
+    target_adaptor: TargetAdaptor,
+    target_type: type[TargetGenerator],
+    generator_fields: dict[str, Any],
+    union_membership: UnionMembership,
+) -> list[tuple[TargetGenerator, dict[str, Any]]]:
+    # Pre-load field values from defaults for the target type being generated.
+    if hasattr(target_type, "generated_target_cls"):
+        family = await Get(AddressFamily, AddressFamilyDir(address.spec_path))
+        template_fields = dict(family.defaults.get(target_type.generated_target_cls.alias, {}))
+    else:
```

Comment:
This is the only new in this function. It had to move before the `@rule` method calling it.